crates/sncast/src/starknet_commands/account/create.rs

```diff
@@ -127,33 +128,34 @@ pub async fn create(
         );
         message.push_str(&deploy_command);
     }
-
-    if add_profile.is_some() {
-        if let Some(url) = &create.rpc.url {
-            let config = CastConfig {
-                url: url.clone(),
-                account: account.into(),
-                accounts_file: accounts_file.into(),
-                keystore,
-                ..Default::default()
-            };
-            add_created_profile_to_configuration(create.add_profile.as_deref(), &config, None)?;
-        } else {
-            unreachable!("Conflicting arguments should be handled in clap");
-        }
+    let mut add_profile_message =
+        String::from("--add-profile flag was not set. No profile added to snfoundry.toml");
+
+    if create.add_profile.is_some() {
+        let url = create
+            .rpc
+            .url
+            .clone()
+            .expect("Conflicting arguments should be handled in clap");
```

Comment:
Nit, this should never even appear, but maybe we can use a more specific message?